coalib/bears/Bear.py

```diff
@@ -22,6 +24,34 @@
 from .meta import bearclass
 
 
+class db(pdb.Pdb):
+    def __init__(self, *args, **kwargs):
+        self.do_q = self.custom_quit
+        self.do_quit = self.custom_quit
+        self.do_exit = self.custom_quit
+        super(db, self).__init__(*args, **kwargs)
+
+    def custom_quit(self, arg):
+        self.clear_all_breaks()
```

Comment:
We keep that, it's definitely good, but I've thought about the scenario of setting breakpoints across bears or utilities. Imagine this:

Some bear fails. You now want to debug. You step into code, experiment around, and you have set some breakpoints. The bear that failed is a local bear, so you step each time a new file arises into `run`. But since `runcall` has ended already for the last `run`, breakpoints you've set are gone.

It would be cool to disable breakpoints on exit, but remember them until the complete coala session ends (or some other point we thing breakpoints shall be kept alive).

But: We do that after merge, that's something to discuss later since it requires more design work.

Reply:
Point noted Bookmarking this comment. :+1:

Reply:
Making sense, I have to take time to think about it,:+1: yes, of course, will discuss  :smiley: